pulsar-common/src/main/java/org/apache/pulsar/client/api/MessageIdAdv.java

```diff
@@ -75,6 +75,8 @@ default int getBatchSize() {
      * @implNote The message IDs of a batch should share a BitSet. For example, given 3 messages in the same batch whose
      * size is 3, all message IDs of them should return "111" (i.e. a BitSet whose size is 3 and all bits are 1). If the
      * 1st message has been acknowledged, the returned BitSet should become "011" (i.e. the 1st bit become 0).
+     * If the caller performs any read or write operations on the return value of this method, they should do so with
+     * lock protection. #22353
```

Comment:
```suggestion
     * lock protection.
```

We don't need to mention a PR number

Reply:
addressed